### ISSUE TITLE
 Homey Pro (Early 2023)

### ISSUE THREAD

**Issue by terryble66** [rickardp/homey-prometheus-exporter#35]
Hi 
Can the app will be available for  Homey Pro (Early 2023) 
Thank you

**Comment by ptahbe** [comment]
Hi,

Yes. Can you please update the app for the new Homey Pro.


Thx

**Comment by Tixhyl** [comment]
This would be awesome!
Thanks

**Comment by rickardp** [comment]
I don’t own a new Homey pro. Maybe if I can get some help figuring out what migration is needed and testing it out. Of course PRs are always welcome

**Comment by Jesa-a1** [comment]
Hi

The prometheus exporter needs an update, as there is a new version for SDKv3 for Homey Pro (early 2023) model.
https://community.homey.app/t/homey-pro-early-2023-app-unsupported-list/70324?page=20

Kind regards
Jan E.

**Comment by ferdw** [comment]
on the Athom webpage is some info on migrating from sdk2 to 3: https://apps.developer.homey.app/upgrade-guides/upgrading-to-sdk-v3

**Comment by ppiixxii** [comment]
I have the homey 2023 on order, hence once received i can assist in testing.

**Comment by Tixhyl** [comment]
The sdk3 also runs on older Homeys by the way, so it will work on your current Homey as well as on the new one

**Comment by eelco2k** [comment]
+1

**Comment by ptahbe** [comment]
Yes. It would be great if this worked on the new Homey Pro.

**Comment by rickardp** [comment]
I followed the migration instructions for v3 and it still works on my Homey and passes validation.

Can someone please verify if it works on the Homey Pro?

https://homey.app/a/io.prometheus/test/

**Comment by Jesa-a1** [comment]
When selecting the early 2023 homey pro, the test version still says not
compatible

[image: image.png]

Kind regards
Jan E.

tor. 6. apr. 2023 kl. 00:02 skrev Rickard ***@***.***>:

**Comment by rickardp** [comment]
I updated the app to port it to V3. It was more or less a one-liner, but I took the opportunity to fix some minor bugs and restructure according to the recommendations in the docs.

New version can be tested here:
https://homey.app/a/io.prometheus/test/

It works on my Homey, but it would be good to have it thoroughly tested on the Homey Pro as I do not have access to this device.

**Comment by rickardp** [comment]
OK, I broke Homey support on the marketplace now. I need to investigate why...

**Comment by Tixhyl** [comment]
It's working fine on my Homey Pro '23 

Nice work!

**Comment by ptahbe** [comment]
Can confirm that the test version works on Homey Pro 2023

**Comment by rickardp** [comment]
There is an open issue currently on timeouts on firmware v10.0.0-rc.72. While a pre release firmware I guess it’s good to fix before releasing this app as v10 probably will have this issue too. Looks like Athom changed some rules for app startup. So no release yet, but if it is working there is no reason you can’t be on the test version until the timeout is figured out

**Comment by rickardp** [comment]
There is a new version published to test (same URL as before, https://homey.app/a/io.prometheus/test/).

it should address some stability issues at startup (due to timeouts), but the most significant change is that I refactored the code to be more maintainable and converted it to TypeScript. The code base has really outgrown the quick hack JavaScript file it once was. With this comes the risk of some regressions, but it should mean less bugs and quicker development in the future.

I’ve been running this version for about half a day and I haven’t seen any isssue this far.

**Comment by Jesa-a1** [comment]
After automatic update of the test version (1.0.0) the app shows is no change in values. And it reports only like 3% of my devices. After a restart of the app, values changes again, but still like only 3% of my devices is reportet. Reboot of homey did not help either.

**Comment by eelco2k** [comment]
I have the same problem that not all devices are getting reported in the /metrics page. (actually only 2 devices)

**Comment by Tixhyl** [comment]
Same for me, homey diagnostics report: 9ab4947e-0aa0-4bd8-9c9b-4847ae3ea36a

Op za 8 apr. 2023 10:27 schreef Eelco ***@***.***>:

**Comment by rickardp** [comment]
Yes, I managed to introduce a last minute bug. Version 1.0.1 should fix this.

**Comment by rickardp** [comment]
If you experience issues with 1.0.1, it would be much appreciated if you could send a diagnostic report. I've seen some timeout issues still, and there might be a need to add further retry/defer logic. I think some API calls are very slow in some circumstances but I only see the stack trace, which is useless in async code. Hopefully a diagnostic report would help shed some light on this issue.

**Comment by rickardp** [comment]
1.0.2 published with an init crash due to a number variable having a non-number value.

**Comment by rickardp** [comment]
Version 1.0.2 is now live. No issues reported so far.

Closing as completed. For any new issues found, please open a new issue.

Thank you all for helping out with testing!
